src/nvim/api/buffer.c

```diff
@@ -411,10 +411,13 @@ void nvim_buf_set_lines(uint64_t channel_id,
 
 /// Gets a buffer-scoped (b:) variable.
 ///
-/// @param buffer     Buffer handle
-/// @param name       Variable name
-/// @param[out] err   Error details, if any
-/// @return Variable value
+/// @note Use buffer_get_changedtick() for “variable” `b:changedtick`.
+///       This method only gets variables from b: dictionary.
```

Comment:
No, I think that `b:changedtick` should be deprecated. This is a *unique* variable with some hacks around it, normally such internals are exposed via functions (e.g. window identifiers are exposed via `win_getid()`, not via `w:` variable). If or when #243 proceeds most of VimL functions like this will be separate API calls.